cmd/zpool/zpool_main.c

```diff
@@ -479,6 +505,97 @@ usage(boolean_t requested)
 	exit(requested ? 0 : 2);
 }
 
+/*
+ * zpool initialize [-cs] <pool> [<vdev> ...]
+ * Initialize all unused blocks in the specified vdevs, or all vdevs in the pool
+ * if none specified.
+ *
+ *	-c	Cancel. Ends active initializing.
+ *	-s	Suspend. Initializing can then be restarted with no flags.
+ */
+int
+zpool_do_initialize(int argc, char **argv)
+{
+	int c;
+	char *poolname;
+	zpool_handle_t *zhp;
+	nvlist_t *vdevs;
+	int err = 0;
+
+	struct option long_options[] = {
+		{"cancel",	no_argument,		NULL, 'c'},
+		{"suspend",	no_argument,		NULL, 's'},
+		{0, 0, 0, 0}
+	};
+
+	pool_initialize_func_t cmd_type = POOL_INITIALIZE_DO;
+	while ((c = getopt_long(argc, argv, "cs", long_options, NULL)) != -1) {
+		switch (c) {
+		case 'c':
+			if (cmd_type != POOL_INITIALIZE_DO) {
+				(void) fprintf(stderr, gettext("-c cannot be "
```

Comment:
nit: unlike other zpool commands this does not seem to handle duplicate options

```
root@linux:~# zpool initialize testpool
root@linux:~# zpool initialize -s -s testpool
-s cannot be combined with other options
usage:
	initialize [-cs] <pool> [<device> ...]
root@linux:~# zpool initialize -c -c testpool
-c cannot be combined with other options
usage:
	initialize [-cs] <pool> [<device> ...]
root@linux:~# zpool scrub -s -s testpool
cannot cancel scrubbing testpool: there is no active scrub
root@linux:~# zpool scrub -p -p testpool
cannot pause scrubbing testpool: there is no active scrub
```

Also the synopsis seems to suggest both `-c` and `-s` can be used simultaneously, maybe something like `[ -c | -s ]` instead of `[-cs]` would be better.

Reply:
Fixed